assignments/array-methods.js

```diff
@@ -56,27 +56,46 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(runner) {
+ return fullName.push(runner["first_name"] + " " + runner["last_name"]);
+});
 console.log(fullName);
 
+
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = runners.map( function(item) {
+ return item.first_name.toUpperCase();
+});
+console.log(allCaps);
+
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter((size) =>{
+ return size.shirt_size === "L";
+});
 console.log(largeShirts);
 
+
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+const charitableDonas = runners.reduce((theReducer, item) => { 
+    return theReducer += item.donation;},0);
+    ticketPriceTotal.push(charitableDonas);
 console.log(ticketPriceTotal);
 
+
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// The race director needs all companies capitalized for banners...
+let allCaps = runners.map( function(item) {
+ return item.company_name.toUpperCase();
```

Comment:
You created two variables with the same name (you used allCaps above) so your program threw an error.